### PR TITLE
Feature/advnanced search prescriptions

### PR DESCRIPTION
**Pull request by Stefann11**

### CODE REVIEW

File: PatientWebApplication/HealthClinic.DAL/Service/PrescriptionService.cs [hunk 1]
```diff
@@ -45,16 +45,154 @@ public List<Prescription> GetPrescriptionsForPatient(int idPatient)
...
+             
+                if (i == 0)
+                {
+                    finalPrescriptions = searchForLogicOperators(dto.LogicOperators[i], othersPrescriptions, firstPrescriptions);
+                }
+                else
+                {
+                    finalPrescriptions = searchForLogicOperators(dto.LogicOperators[i], othersPrescriptions, finalPrescriptions);
+                }
+            }
```
Comment (Nemanja0505) [inline PatientWebApplication/HealthClinic.DAL/Service/PrescriptionService.cs:85]: Why if and else do the same thing? That needs to be corrected .

File: PatientWebApplication/HealthClinic.DAL/Service/PrescriptionService.cs [hunk 2]
```diff
@@ -45,16 +45,154 @@ public List<Prescription> GetPrescriptionsForPatient(int idPatient)
...
+        /// </param>
+        /// <returns> List of filtered patient prescriptions. </returns>
+        private List<Prescription> searchForDoctor(List<Prescription> prescriptions, PrescriptionSearchDto prescriptionSearchDto)
+        {
+            if (!prescriptionSearchDto.Doctor.Equals(""))
+            {
+                prescriptions = prescriptions.FindAll(prescription => prescription.Doctor.firstName.Contains(prescriptionSearchDto.Doctor) || prescription.Doctor.secondName.Contains(prescriptionSearchDto.Doctor) || prescription.Doctor.DoctorFullName().Contains(prescriptionSearchDto.Doctor));
+            }
+
+            return prescriptions;
```
Comment (Nemanja0505) [inline PatientWebApplication/HealthClinic.DAL/Service/PrescriptionService.cs:195]: For each condition where it is checked whether a string is empty use the method implemented for that check within the utility methods class .

**Review by Nemanja0505 — CHANGES_REQUESTED** [review]
PR cannot be approved elements of  clean code are not following.

**Review by Nemanja0505 — APPROVED** [review]
PR can be approve, clean code is following .

File: PatientWebApplication/HealthClinic.DAL/Service/PrescriptionService.cs [hunk 3]
```diff
@@ -45,16 +46,145 @@ public List<Prescription> GetPrescriptionsForPatient(int idPatient)
...
+        private List<Prescription> searchForLogicOperators(string logicOperator, List<Prescription> othersPrescriptions, List<Prescription> finalPrescriptions)
+        {
+            if (logicOperator.Equals("or"))
+            {
+                return othersPrescriptions.Union(finalPrescriptions).ToList();
+            }
+            else
+            {
+                return othersPrescriptions.Intersect(finalPrescriptions).ToList();
+            }
```
Comment (trki293) [inline PatientWebApplication/HealthClinic.DAL/Service/PrescriptionService.cs:91]: Write the highlighted code sequence in the form of a ternary operator.

File: PatientWebApplication/HealthClinic.DAL/Service/PrescriptionService.cs [hunk 4]
```diff
@@ -45,16 +46,145 @@ public List<Prescription> GetPrescriptionsForPatient(int idPatient)
...
+                return searchForCommentsAdvanced(prescriptions, otherValue);
+            }
+            else if (otherParameter.Equals("isUsed"))
+            {
+                return searchForUsedAdvanced(prescriptions, otherValue);
+            }
+            else
+            {
+                return searchForDoctorAdvanced(prescriptions, otherValue);
+            }
```
Comment (trki293) [inline PatientWebApplication/HealthClinic.DAL/Service/PrescriptionService.cs:111]: Write the highlighted code sequence in the form of a ternary operator.

File: PatientWebApplication/HealthClinic.DAL/Service/PrescriptionService.cs [hunk 5]
```diff
@@ -45,16 +46,145 @@ public List<Prescription> GetPrescriptionsForPatient(int idPatient)
...
+                return searchForCommentsAdvanced(prescriptions, dto.First);
+            }
+            else if (dto.FirstRole.Equals("isUsed"))
+            {
+                return searchForUsedAdvanced(prescriptions, dto.First);
+            }
+            else
+            {
+                return searchForDoctorAdvanced(prescriptions, dto.First);
+            }
```
Comment (trki293) [inline PatientWebApplication/HealthClinic.DAL/Service/PrescriptionService.cs:131]: Write the highlighted code sequence in the form of a ternary operator.

**Review by trki293 — CHANGES_REQUESTED** [review]
Correct the highlighted code to meet good Clean code practices.

**Review by trki293 — APPROVED** [review]
All indicated places in the code have been corrected, the principles of Clean code have been followed. PR is approved.

**Pull request merged**